src/network.ts

```diff
@@ -1,27 +1,64 @@
-import { Handle } from "./core/contracts/registry";
-import { ConfigOpts } from "./config";
-import { NotImplementedError } from "./core/utilities/errors";
+import { resolveRegistration, Handle } from "./core/contracts/registry";
+import * as messages from "./core/messages/messages";
+import * as batchMessages from "./core/batch/batchMesssages";
+import * as config from "./config";
+import { UserNotFound } from "./handles";
+import { MissingUser, NotImplementedError } from "./core/utilities";
 
 /**
- * follow() creates a follow event and enqueues it for the next batch. This
- * method is not yet implemented.
+ * follow() creates a follow event and returns it.
  *
- * @param handle  The handle of the user to follow
- * @param opts    Optional. Configuration overrides, such as from address, if any
+ * @param handle The handle of the user to follow
+ * @param opts   Optional. Configuration overrides, such as from address, if any
+ * @returns      The signed DSNP Graph Change message
  */
-export const follow = async (_handle: Handle, _opts?: ConfigOpts): Promise<void> => {
-  throw NotImplementedError;
+export const follow = async (
+  handle: Handle,
+  opts?: config.ConfigOpts
+): Promise<batchMessages.BatchGraphChangeMessage> => {
+  // Get current user id
+  const { currentUserId } = await config.getConfig(opts);
```

Comment:
There's lot of this same checking in several places and it's also resulted in a lot of the same tests. I think this could use a refactoring pass.  Maybe for a follow up PR.  It's obvious we should never continue with creating messages without all the config pieces in place.

Reply:
@wilwade made a good suggestion a few days ago about adding some methods like `getSigner()` and `getStore()` to the config module. This bit would fit well in a `getCurrentFromId()` method in the same place.

I'll throw a chore on the backlog for us to discuss in the next planning meeting.

Reply:
https://www.pivotaltracker.com/story/show/178458945